### PR TITLE
Improve documentation for MultiChildLayoutDelegate

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: packages/flutter/lib/src/rendering/custom_layout.dart [hunk 1]
```diff
@@ -148,14 +166,32 @@ abstract class MultiChildLayoutDelegate {
...
+  /// the final position of each child with [positionChild].
   void performLayout(Size size, BoxConstraints constraints);
 
+  /// Override this method to return true when the children need to be
+  /// laid out. This should compare the fields of the current delegate
+  /// and the given oldDelegate and return true if the fields are such
+  /// that the layout would be different.
+  bool shouldRelayout(MultiChildLayoutDelegate oldDelegate);
+
+  /// Override this to include additional information in the debugging
```
Comment (HansMuller) [inline packages/flutter/lib/src/rendering/custom_layout.dart]: override this method

File: packages/flutter/lib/src/rendering/custom_layout.dart [hunk 2]
```diff
@@ -15,21 +16,30 @@ class MultiChildLayoutParentData extends ContainerBoxParentDataMixin<RenderBox>
 }
 
 /// A delegate that controls the layout of multiple children.
+///
+/// Used with [MultiChildCustomLayout], the widget for the
+/// [RenderCustomMultiChildLayoutBox] render object.
+///
+/// Subclass this object and override some or all of the methods
```
Comment (HansMuller) [inline packages/flutter/lib/src/rendering/custom_layout.dart]: Subclass this object sounds a little obtuse. Maybe just say "Subclasses must override some or ..."

**Comment by HansMuller** [comment]
LGTM

**Comment by Hixie** [comment]
Fixed, thanks. Will check in when travis goes green.

**Pull request merged**